rules/plists.bzl

```diff
@@ -1,6 +1,92 @@
+"""
+Defines macros for working with plist files.
+"""
+
 load("@bazel_skylib//lib:types.bzl", "types")
+load("@bazel_skylib//lib:sets.bzl", "sets")
+load("@build_bazel_rules_ios//rules:substitute_build_settings.bzl", "substitute_build_settings")
 load("//rules:library.bzl", "write_file")
-load("//rules/library:xcconfig.bzl", "build_setting_name")
+load("//rules/library:xcconfig.bzl", "build_setting_name", "merge_xcconfigs")
+
+def process_infoplists(name, infoplists, infoplists_by_build_setting, xcconfig, xcconfig_by_build_setting):
+    """
+    Constructs substituted_plists by substituting build settings from an xcconfig dict into the variables of a plist.
+
+    Args:
+        name: The name of the target that the plist is being generated for.
+        infoplists: The plist files to be manipulated
+        infoplists_by_build_setting: A dictionary of infoplists keyed by config_setting, merged with infoplists created here.
+        xcconfig: the default build settings to expand the infoplists with
+        xcconfig_by_build_setting: the build settings grouped by config_setting to expand the infoplists with
+
+    Returns:
+        A selectable dict of the substituted_plists grouped by config_setting
+    """
+    infoplists_by_build_setting = dict(infoplists_by_build_setting)
+    xcconfig_by_build_setting = dict(xcconfig_by_build_setting)
+
+    # Final substituted infoplists_by_build_setting dict
+    substituted_infoplists_by_build_setting = {}
+
+    # Default infoplists are any defined in `infoplists` OR in `infoplists_by_build_setting` with the key `//conditions:default`
+    default_infoplists = infoplists_by_build_setting.pop("//conditions:default", infoplists)
```

Comment:
@luispadron is it fair to say that this logic just discards `infoplists` if you provide it with an `info_plists_by_build_setting["//conditions:default"] != None`?

It looks like it might be a breaking change / is divergent from this API is working in other contexts:
https://github.com/bazel-ios/rules_ios/blob/master/rules/library.bzl#L469

Reply:
Yes thats what it'll do but thats the same way it worked before my changes: https://github.com/bazel-ios/rules_ios/blob/9c621595602e2cd6bdda711a20fba1c5e0bc2fd2/rules/plists.bzl#L41

The `infoplists_by_build_setting` and `xcconfig_by_build_setting` are local copies so `pop` shouldn't affect downstream

Reply:
Ok thanks for clarifying 🤯